src/vm/types/execution_scopes.zig

```diff
@@ -5,9 +5,27 @@ const Felt252 = @import("../../math/fields/starknet.zig").Felt252;
 const HintError = @import("../error.zig").HintError;
 
 pub const HintType = union(enum) {
+    const Self = @This();
     // TODO: Add missing types
     felt: Felt252,
     u64: u64,
+    felt_map_of_arr_u64: std.AutoHashMap(Felt252, std.ArrayList(u64)),
+    arr_u64: std.ArrayList(u64),
+
+    pub fn deinit(self: *Self) void {
+        switch (self.*) {
+            .felt_map_of_arr_u64 => |*d| {
+                var it = d.valueIterator();
+                while (it.next()) |v| v.deinit();
+
+                d.deinit();
+            },
+            .arr_u64 => |d| {
+                d.deinit();
+            },
+            else => {},
+        }
+    }
```

Comment:
Is it working? Because here deinit values are passed by values but should be passed as pointers to be effective no?

Like for example in:

```zig
.arr_u64 => |d| {
    d.deinit();
},
```

`d` is passed by value here no?

Reply:
arraylist deinit accept by value not ref, so i think if we here pass by ref, we will just overhead